test/gtest/soft_max.cpp

```diff
@@ -55,34 +44,138 @@ T logaddexp(T x, T y, T neg_inf)
+    in_dim_set.erase({1, 480, 128, 256});
+
+    /// \todo Resolve this workaround. Regular failures on Radeon VII, ROCm 3.3:
+    /// --float --input-dim 1 1 8 8 --algorithm 0 --mode 1 --scales 1 0 --tolerance 8000
+    /// FAILED: -nan
+    in_dim_set.erase({1, 1, 8, 8});
+    in_dim_set.erase({1, 1, 14, 14});
+    in_dim_set.erase({1, 1, 27, 27});
+    in_dim_set.erase({1, 32, 7, 7});
+    in_dim_set.erase({1, 32, 8, 8});
+
+    std::vector<int> algos                 = {0, 1, 2};
+    std::vector<int> modes                 = {0, 1};
+    std::vector<std::vector<float>> scales = {{1.0f, 0.0f}, {0.5f, 0.5f}};
+
+    std::vector<TestCase> test_cases;
+
+    for(const auto& in_dim : in_dim_set)
+        for(const int algo : algos)
+            for(const int mode : modes)
+            {
+                AddTestCasesForDifferentScales<T>(test_cases, in_dim, algo, mode, scales);
+            }
+
+    return test_cases;
+}
+
+template <typename T>
+auto GetCases()
+{
+    static const auto cases = testing::ValuesIn(GenCases<T>());
+    return cases;
+}
+
+} // namespace
+
+template <typename T>
+struct SoftmaxCommon : public testing::TestWithParam<TestCase>
+{
+    void SetUp() override { prng::reset_seed(); }
+
+    void Run()
+    {
+        const TestCase& test_case = GetParam();
+
+        uint64_t max_value =
+            miopen_type<T>{} == miopenHalf ? (test_case.algo == MIOPEN_SOFTMAX_LOG ? 3 : 5) : 17;
+
+        input            = tensor<T>{test_case.in_dim}.generate(tensor_elem_gen_integer{max_value});
+        size_t total_mem = 2 * input.desc.GetNumBytes(); // estimate based on backward pass
+        size_t device_mem = get_handle().GetGlobalMemorySize();
+        if(total_mem >= device_mem)
+        {
+            std::cout << "Config requires " << total_mem
+                      << " Bytes to write all necessary tensors to GPU. GPU has " << device_mem
+                      << " Bytes of memory." << std::endl;
+
+            GTEST_SKIP();
+        }
+
+        output = tensor<T>{test_case.in_dim}.generate(tensor_elem_gen_integer{max_value});
+
+        std::vector<T> tensorCpuDataForward = GetForwardCpu();
+        std::vector<T> tensorGpuDataForward = GetForwardGpu();
+
+        // check forward results
+        CompareResults(tensorGpuDataForward, tensorCpuDataForward);
+
+        dout   = tensor<T>{test_case.in_dim}.generate([&](int n, int c, int h, int w) {
+            T x      = input(n, c, h, w);
+            double y = (877 * n + 547 * c + 701 * h + 1049 * w + static_cast<int>(769 * x)) % 2503;
+            return ((x * y) / 1301.0);
+        });
+        dinput = tensor<T>{test_case.in_dim}.generate(tensor_elem_gen_integer{max_value});
+
+        std::vector<T> tensorCpuDataBackward = GetBackwardCpu();
+        std::vector<T> tensorGpuDataBackward = GetBackwardGpu();
+
+        // check backward results
+        CompareResults(tensorGpuDataBackward, tensorCpuDataBackward);
+    }
+
+    std::vector<T> GetForwardCpu() const
```

Comment:
we also have [mloSoftmaxForwardRunHost ](https://github.com/ROCm/MIOpen/blob/8be8d80c3ab6567fa0947cb59ee6943358557d93/driver/mloSoftmaxHost.hpp#L49) and [mloSoftmaxBackwardRunHost ](https://github.com/ROCm/MIOpen/blob/8be8d80c3ab6567fa0947cb59ee6943358557d93/driver/mloSoftmaxHost.hpp#L243)too.

Reply:
hi, @bghimireamd 
In C-test I did not see calls of these functions. Scope of my refactoring was to replace c-test with g-test.
If I misunderstand something, lets discuss.

Reply:
Or do you mean that we need to enhance test coverage for softmax? If yes I would prefer to do it in separate PR/ticket.

Reply:
yeah separate PR/ticket make more sense here :)